### ISSUE TITLE
Modality stacking inversion in View->Goto error

### ISSUE THREAD

**Issue by nabijaczleweli** [fontforge/fontforge#5493]
### When reporting a bug/issue:
- [x] Screenshot: ![image](https://github.com/user-attachments/assets/c2f8bd11-2e3e-43ff-80e4-529728b60008)
- [x] The FontForge version and the operating system you're using: 5818fb88035f451b5d72182dee8228e5ecf10d29 and bookworm 1:20230101~dfsg-1.1~deb12u1
- [x] The behavior you expect to see:
      * the error window is clickable, and can be OKed away or closed
- [x] and the actual behavior:
      * the "Enter the glyph name..." window *remains modal*
      * you can only click the "Enter the glyph name..." window. you need to close it before dismissing the error, and then you need to reopen it to search again
- [x] Steps to reproduce the behavior: View->Goto, aslkhfjkashfasklfh, OK, click OK (or try to close) the error window (doesn't work), close the Goto window, close the error window

**Comment by nabijaczleweli** [comment]
Honestly maybe Goto just shouldn't be modal? It still works if you keep typing under the error. Of course, `ff_post_notice()` will only ever show one notice, so multiple failed gotos won't show multiple errors (bad). idk

**Comment by iorsh** [comment]
Non-modal windows tend to introduce complications. All sorts of data syncing, what to do when the window is opened multiple times, etc. Some windows have a good reason to be non-modal, but not this one, I think.

**Comment by iorsh** [comment]
I noticed that the first error goes to the error window, and other errors go to the non-modal "Warnings" window where all the errors are accumulated. So this could be a general issue with warnings mechanism and not with the "Goto" dialog.

**Comment by nabijaczleweli** [comment]
I actually thought all the other errors were swallowed somehow tbh. Now that you mention I do see a "Warnings" window in the tab order but it never renders anywhere.
